tests/deliver_to_nonexistent_contract.go

```diff
@@ -0,0 +1,146 @@
+	Expect(err).Should(BeNil())
+	destinationContractAddress, err := deploymentUtils.DeriveEVMContractAddress(deployerAddress, nonce)
+	Expect(err).Should(BeNil())
+
+	//
+	// Call the example messenger contract on Subnet A
+	//
+	message := "Hello, world!"
+	optsA = utils.CreateTransactorOpts(ctx, subnetAInfo, fundedAddress, fundedKey)
+	tx, err = subnetAExampleMessenger.SendMessage(optsA, subnetBInfo.BlockchainID, destinationContractAddress, fundedAddress, big.NewInt(0), big.NewInt(300000), message)
+	Expect(err).Should(BeNil())
+
+	// Wait for the transaction to be mined
+	receipt, err = bind.WaitMined(ctx, subnetAInfo.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	sendEvent, err := utils.GetSendEventFromLogs(receipt.Logs, subnetATeleporterMessenger)
+	Expect(err).Should(BeNil())
+	Expect(sendEvent.DestinationChainID[:]).Should(Equal(subnetBInfo.BlockchainID[:]))
+
+	teleporterMessageID = sendEvent.Message.MessageID
+
+	//
+	// Relay the message to the destination
+	//
+
+	receipt = network.RelayMessage(ctx, receipt, subnetAInfo, subnetBInfo, true)
+	receiveEvent, err := utils.GetReceiveEventFromLogs(receipt.Logs, subnetATeleporterMessenger)
+	Expect(err).Should(BeNil())
+	deliveredTeleporterMessage := receiveEvent.Message
+
+	//
+	// Check that the message was successfully relayed
+	//
+	delivered, err := subnetBTeleporterMessenger.MessageReceived(&bind.CallOpts{}, subnetAInfo.BlockchainID, teleporterMessageID)
+	Expect(err).Should(BeNil())
+	Expect(delivered).Should(BeTrue())
+
+	//
+	// Check that the message was not successfully executed
+	//
+	executionFailedEvent, err := utils.GetMessageExecutionFailedFromLogs(receipt.Logs, subnetBTeleporterMessenger)
+	Expect(err).Should(BeNil())
+	Expect(executionFailedEvent.Message.MessageID).Should(Equal(deliveredTeleporterMessage.MessageID))
+
+	//
+	// Deploy the contract on Subnet B
+	//
+	optsB := utils.CreateTransactorOpts(ctx, subnetBInfo, deployerAddress, deployerKey)
+	exampleMessengerContractB, tx, subnetBExampleMessenger, err := examplecrosschainmessenger.DeployExampleCrossChainMessenger(optsB, subnetBInfo.ChainRPCClient, subnetBInfo.TeleporterRegistryAddress)
+
+	// Wait for the transaction to be mined
+	_, err = bind.WaitMined(ctx, subnetBInfo.ChainRPCClient, tx)
+	Expect(err).Should(BeNil())
+	Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+	// Confirm that it was deployed at the expected address
+	Expect(exampleMessengerContractB).Should(Equal(destinationContractAddress))
+
+	//
+	// Call retryMessageExecution on Subnet B
+	//
+	signedTx := utils.CreateRetryMessageExecutionTransaction(
+		ctx,
+		subnetBInfo,
+		subnetAInfo.BlockchainID,
+		deliveredTeleporterMessage,
+		fundedAddress,
+		fundedKey,
+		teleporterContractAddress,
+	)
+	utils.SendTransactionAndWaitForAcceptance(ctx, subnetBInfo.ChainWSClient, subnetBInfo.ChainRPCClient, signedTx, true)
```

Comment:
do we want to use abi binding to call `RetryMessageExecution` directly?

Reply:
Good call, fixed now